travis.sh

```diff
@@ -179,25 +181,32 @@ function prepare_ros_workspace() {
          http://* | https://* | file://*) ;; # use url as is
          *) item="file://$CI_SOURCE_PATH/$item" ;; # turn into proper url
       esac
-      travis_run_true wstool merge -k $item
+      travis_run_true curl -s -o $upstream_workspace_file $item
       test $? -ne 0 && echo -e "$(colorize RED Failed to find rosinstall file. Aborting.)" && exit 2
    done
 
-   # Download upstream packages into workspace
-   if [ -e .rosinstall ]; then
-      # ensure that the to-be-tested package is not in .rosinstall
-      travis_run_true wstool rm $REPOSITORY_NAME
-      # perform shallow checkout: only possible with wstool init
-      travis_run_simple mv .rosinstall rosinstall
-      travis_run cat rosinstall
-      travis_run wstool init --shallow . rosinstall
-   fi
-
    # Link in the repo we are testing
    if [ "$(dirname $CI_SOURCE_PATH)" != $PWD ] ; then
       travis_run_simple --title "Symlinking to-be-tested repo $CI_SOURCE_PATH into ROS workspace" ln -s $CI_SOURCE_PATH .
    fi
 
+   # Download upstream packages into workspace
+   if [ -e $upstream_workspace_file ]; then
+      travis_run cat $upstream_workspace_file
+      # Clone all package dependencies into subfolder ./upstream/ to prevent name conflicts.
+      # The flag '--skip-existing' whill ignore packages with the same repo url as the test package.
+      upstream_folder=upstream
+      travis_run mkdir $upstream_folder
+      travis_run vcs import --skip-existing --input $upstream_workspace_file $upstream_folder
```

Comment:
Where is `vcs` installed?

Reply:
Does vcs support shallow clones?

Reply:
Just as `wstool`, `vcs` is provided by the corresponding docker container.
It's possible to pass custom git arguments and to enable shallow clones by doing so.

Reply:
Would be great to do so then.

Reply:
Apparently, custom git commands are only possible for already cloned packages. I'll see if we can get this feature added to vcstool